docs/java-rest/high-level/apis/scroll.asciidoc

```diff
@@ -0,0 +1,157 @@
+[[java-rest-high-search-scroll]]
+=== Search Scroll API
+
+The Scroll API can be used to retrieve a large number of results from
+a search request.
+
+In order to use scrolling, the initial search request must define
+a value for the scroll parameter:
+
+["source","java",subs="attributes,callouts,macros"]
+--------------------------------------------------
+include-tagged::{doc-tests}/SearchDocumentationIT.java[search-scroll-request]
+--------------------------------------------------
+<1> Create a new `SearchRequest`. See <<java-rest-high-search>>
+for more information on how to build `SearchRequest`.
+<2> Set the `scroll` parameter as a `TimeValue` corresponding to one minute
+
+When executing the `SearchRequest`, Elasticsearch detects the presence
+of the `scroll` parameter and keeps the search context alive during the time
+defined by the parameter.
+
+It then returns a `SearchResponse` that includes a scroll id:
+
+["source","java",subs="attributes,callouts,macros"]
+--------------------------------------------------
+include-tagged::{doc-tests}/SearchDocumentationIT.java[search-response-scroll-id]
+--------------------------------------------------
+<1> Execute the `SearchRequest`
+<2> Retrieve the search hits
+<3> Retrieve the scroll id
+
+This scroll id should be passed to a `SearchScrollRequest`:
+
+["source","java",subs="attributes,callouts,macros"]
+--------------------------------------------------
+include-tagged::{doc-tests}/SearchDocumentationIT.java[search-scroll-id]
+--------------------------------------------------
+<1> Create a new `SearchScrollRequest`
+<2> Set the `scroll` parameter again to tell Elasticsearch to keep the search context
+alive for another minute
+<3> Set the scroll id
+
+And the `SearchScrollRequest` can be executed using the Search Scroll API in order
+to retrieve the next batch of results:
+
+["source","java",subs="attributes,callouts,macros"]
+--------------------------------------------------
+include-tagged::{doc-tests}/SearchDocumentationIT.java[search-scroll-execute]
+--------------------------------------------------
+<1> Execute the `SearchScrollRequest`
+<2> Retrieve the next batch of search hits
+<3> Retrieve the next scroll id to use in upcoming requests
+
+Then the same process can be repeated over and over until no more results are
+returned. The initial search request as well as all subsequent scroll requests
+return a scroll id that can be passed to the next scroll request until results
+are exhausted.
+
+Finally, the scroll id can be deleted using the <<java-rest-high-clear-scroll>>.
+
+==== Optional arguments
+The following arguments can optionally be provided:
+
+["source","java",subs="attributes,callouts,macros"]
+--------------------------------------------------
+include-tagged::{doc-tests}/SearchDocumentationIT.java[scroll-request-scroll]
+--------------------------------------------------
+<1> Scroll value (ie, the time to keep alive the search context) as a `TimeValue`
```

Comment:
I didn't realize that the scroll was optional. Maybe we should mention the default value when not specified?